filewatcher/filewatcher.go

```diff
@@ -72,34 +64,27 @@ type Parser func(f io.Reader) (data any, err error)
 // as does returning nil data and nil error.
 //
 // Use WrapDirParser to wrap it into a Parser to be used with FileWatcher.
-type DirParser func(dir fs.FS) (data any, err error)
+type DirParser = v2.DirParser[any]
 
 // WrapDirParser wraps a DirParser for a directory to a Parser.
 //
 // When using FileWatcher to watch a directory instead of a single file,
 // you MUST use WrapDirParser instead of any other Parser implementations.
 func WrapDirParser(dp DirParser) Parser {
-	return func(r io.Reader) (data any, err error) {
-		path := string(r.(fakeDirectoryReader))
-		dir := os.DirFS(path)
-		return dp(dir)
-	}
+	return v2.WrapDirParser(dp)
 }
 
 // Result is the return type of New. Use Get function to get the actual data.
 type Result struct {
-	data atomic.Value
-
-	ctx    context.Context
-	cancel context.CancelFunc
+	result *v2.Result[any]
```

Comment:
🔕 value rather than pointer?

Reply:
hmm, it recognizes value types of `sync.Mutex` as long as you don't copy the containing struct... this doesn't seem like it should be different.  Can you create the struct without copying?

Reply:
non-pointer type copy-protected fields are appropriate when you just use their zero values in the constructors directly (e.g. locks, zero atomic values, etc.), for this, since it's a wrapper from v2, when constructing it you need to get the data from v2 and feed into this type, so a copy will always happen.